flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveSourceDynamicFileEnumerator.java

```diff
@@ -131,6 +132,7 @@ RowData createRowData(RowType rowType, Map<String, String> partitionSpec) {
         GenericRowData rowData = new GenericRowData(rowType.getFieldCount());
         for (int i = 0; i < rowType.getFieldCount(); ++i) {
             String value = partitionSpec.get(dynamicFilterPartitionKeys.get(i));
+            LogicalType fieldType = rowType.getTypeAt(i);
             Object convertedValue =
```

Comment:
call `rowData.setField(i, null);` and continue ?

Reply:
I'd suggest not to do the short-cut. The inner type conversions support converting null values, and in case any of the conversions converts a non-null value to a null value, which may lead to NPE, I think it's better set/skip the null values after the inner conversions are done.